storage/client.go

```diff
@@ -48,6 +49,10 @@ const (
 	userAgentHeader = "User-Agent"
 )
 
+var (
+	validStorageAccount = regexp.MustCompile("^[0-9a-z]{3,24}$")
```

Comment:
I'm concerned about having this logic in our SDK.  If/when this restriction changes we will need to update our logic too.  Is there a way we can have Azure validate the name?